### PR TITLE
REF: Fix PeriodIndex._outer_indexer -> share remaining set methods

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/datetimelike.py [hunk 1]
```diff
@@ -837,13 +844,6 @@ def _union(self, other, sort):
     # Join Methods
     _join_precedence = 10
 
-    _inner_indexer = _join_i8_wrapper(libjoin.inner_join_indexer)
```
Comment (jreback) [inline pandas/core/indexes/datetimelike.py:840]: I think remove L836 & 837 and move L838 to same location makes sense

File: pandas/tests/indexes/period/test_join.py [hunk 2]
```diff
@@ -8,6 +8,12 @@
 
 
 class TestJoin:
+    def test_join_outer_indexer(self):
+        # smoke check that PeriodIgit standex._outer_indexer works
+        pi = period_range("1/1/2000", "1/20/2000", freq="D")
+
+        pi._outer_indexer(pi._values, pi._values)
```
Comment (jreback) [inline pandas/tests/indexes/period/test_join.py:15]: can you assert the result

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/datetimelike.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/datetimelike.py:840]: Tried that, moving 838 breaks things bc it changes PeriodIndex._join_precedence

File: pandas/tests/indexes/period/test_join.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/indexes/period/test_join.py:15]: sure, will update

Comment (jbrockmendel) [reply pandas/tests/indexes/period/test_join.py]: updated + green

**Review by jreback — APPROVED** [review]

**Pull request merged**